### PR TITLE
doc: add support status page - v4

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: #9163 

Changes from last PR:
- Incorporate feedback.

Generated at https://jasonish-suricata.readthedocs.io/en/doc-support-status-v4/support-status.html.

### CODE REVIEW

File: doc/userguide/support-status.rst [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+Tier 1 supported items are developed and supported by the Suricata
+core team. These items receive full CI (continuous integration)
+coverage, and functional failures block git merges and releases. Tier
+1 features are enabled by default on platforms that support the
+feature.
+
+Tier 2
+------
+
+Tier 2 supported items are developed and support by the Suricata core
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:24]:
```suggestion
Tier 2 supported items are developed and supported by the Suricata core
```

File: doc/userguide/support-status.rst [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+team, sometimes with help from community. Major functional failures
+block git merges and releases, however less major issues may be
+documented as "known issues" and may go into a release. Tier 1
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:27]:
```suggestion
documented as "known issues" and may go into a release. Tier 2
```

File: doc/userguide/support-status.rst [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+core team. These items receive full CI (continuous integration)
+coverage, and functional failures block git merges and releases. Tier
+1 features are enabled by default on platforms that support the
+feature.
+
+Tier 2
+------
+
+Tier 2 supported items are developed and support by the Suricata core
+team, sometimes with help from community. Major functional failures
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:25]:
```suggestion
team, sometimes with help from the community. Major functional failures
```

File: doc/userguide/support-status.rst [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+avoid overloading the team.
+
+When accepting a feature into the code base anyway, it will come with
+a number of limits and conditions:
+
+* submitter must commit to maintaining it:
+
+  - make sure code compiles and correctly functions after Suricata
+    and/or external (e.g. library) changes.
+  - support users when they encounter problems on the mailinglists and
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:44]:
```suggestion
  - support users when they encounter problems on the forum and
```

Comment (jufajardini) [reply doc/userguide/support-status.rst:44]: Should we add Discord? Sounds too much, considering how volatile messages there can be.

File: doc/userguide/support-status.rst [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    redmine tickets
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:45]: nit:
```suggestion
    redmine tickets.
```

File: doc/userguide/support-status.rst [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+* the code will be disabled by default and will not become part of the
+  QA setup. This means it will be enabled only by an ``--enable``
+  configure flag.
+
+* the code may not have CI coverage by the OISF infrastructure
```
Comment (jufajardini) [inline doc/userguide/support-status.rst:51]: nit:
```suggestion
* the code may not have CI coverage by the OISF infrastructure.
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
Some final nits :P

**Comment by victorjulien** [comment]
included in #9209

**Pull request closed without merging**